contracts/QuestNFT.sol

```diff
@@ -90,19 +71,54 @@ contract QuestNFT is Initializable, ERC721Upgradeable, PausableUpgradeable, Owna
         _;
     }
 
-    /// @dev set jsonSpecCID only if its empty
-    /// @param jsonSpecCID_ The jsonSpecCID to set
-    function setJsonSpecCID(string memory jsonSpecCID_) external onlyOwner {
-        require(bytes(jsonSpecCID_).length > 0, 'jsonSpecCID cannot be empty');
-        require(bytes(jsonSpecCID).length == 0, 'jsonSpecCID already set');
+    function addQuest(
+        uint256 questFee_,
+        uint256 startTime_,
+        uint256 endTime_,
+        uint256 totalParticipants_,
+        string memory questId_,
+        string memory description_,
+        string memory imageIPFSHash_
+    ) public onlyMinter returns (uint256) {
+        require (endTime_ > block.timestamp, 'endTime_ in the past');
+        require (startTime_ > block.timestamp, 'startTime_ in the past');
```

Comment:
Should we remove this check? I believe we removed it in `Quest.sol` so that we can start quests immediately